docs/ratt-transform/index.md

```diff
@@ -675,84 +675,384 @@ app.use(
+
+```xml
+<?xml version="1.0"?>
+<root>
+  <metadata>
+    <title>
+      <value>Data about countries.</value>
+    </title>
+  </metadata>
+  <data>
+    <countries>
+      <id>nl</id>
+      <name>The Netherlands</name>
+    </countries>
+    <countries>
+      <id>de</id>
+      <name>Germany</name>
+    </countries>
+  </data>
+</root>
+```
+
+See [the section on connecting XML sources](#xml) for more information.
+
+### Specifying paths (nested keys)
+
+In tabular data, keys (or column names) are singular.  But in tree-shaped data a *path* of the tree can consist of one or more keys that must be traversed in sequence.
+
+Paths are specified as dot-separated sequences of keys, starting at the top-level and ending at the required value.  For the above example, RATT can access the `"name"` key inside the `"title"` key, which itself is nested inside the `"metadata"` key.  This path is expressed in [1].  Notice that the path expressed in [1] is different from the path expressed in [2], which also access the `"name"` key, but nested inside the `"countries"` and then `"data"` keys.
+
+```
+[1] metadata.title.name
+[2] data.countries.name
+```
+
+Path expressions can be used anywhere with singular keys.  For example, we can assert the title of a dataset in the following way:
+
+```ts
+app.use(
+  mw.addQuad(
+    prefix.dataset('my-dataset'),
+    dct.title,
+    mw.toLiteral('metadata.title.value', {language: 'en'})),
+)
+```
+
+This results in the following assertion:
+
+```trig
+dataset:my-dataset dct:title 'Data about countries.'@en.
+```
+
+<h3 id='accessing-lists-by-index'>Accessing lists by index</h3>
+
+Tree-shaped data formats often allow multiple values to be specified in a sequence.  Examples of this are XML tags with the same key that are directly nested under the same parent, and lists in JSON.
+
+RATT is able to access specific elements from such lists based on their *index* or position.  Following the standard practice in Computer Science, RATT refer to the first element in the list as having index 0.  The second element has index 1, etc.
+
+For the above example record, we can assert the first label of the country:
+
+```ts
+app.use(
+  mw.addQuad(
+    mw.toIri('data.countries[0].id', {prefix: prefix.country}),
+    rdfs.label,
+    mw.toLiteral('data.countries[0].name', {language: 'en'})),
+)
+```
+
+This results in the following assertion:
+
+```trig
+country:nl rdfs:label 'The Netherlands'@en.
+```
+
+We can also assert the second label of the same country.  Notice that only the index is different (‘1’ instead of ‘0’):
+
+```ts
+app.use(
+  mw.addQuad(
+    mw.toIri('data.countries[1].id', {prefix: prefix.country}),
+    rdfs.label,
+    mw.toLiteral('data.countries[1].name', {language: 'en'})),
+)
+```
+
+This results in the following assertion:
+
+```trig
+country:de rdfs:label 'Germany'@en.
+```
+
+### Iterating over lists
```

Comment:
Iterating over arrays of objects.

Reply:
I'm going to think about this a bit more.  I don't like having different documentation sections for this.

Reply:
Renamed the section accordingly.